Src/PerformanceCollector/Shared/Implementation/QuickPulse/IQuickPulseServiceClient.cs

```diff
@@ -16,15 +17,33 @@ internal interface IQuickPulseServiceClient
         /// </summary>
         /// <param name="instrumentationKey">InstrumentationKey for which to submit data samples.</param>
         /// <param name="timestamp">Timestamp to pass to the server.</param>
+        /// <param name="configurationETag">Current configuration ETag that the client has.</param>
+        /// <param name="authApiKey">Authentication API key.</param>
+        /// <param name="configurationInfo">When available, the deserialized response data received from the server.</param>
         /// <returns><b>true</b> if data is expected, otherwise <b>false</b>.</returns>
-        bool? Ping(string instrumentationKey, DateTimeOffset timestamp);
+        bool? Ping(
```

Comment:
Looking at this interface the best name would be `TryGetConfiguration`. Since you are hiding an important detail that this method is actually used to transmit `roleInstance` name used for visualization - this renaming would be logical. Please re-think the interface.

Another idea - why wouldn't you initialize the new client whenever you want a new `instrumentationKey`, or `etag` or `authApiKey` to be transmitted?